gatsby/create-pages/create-career-pages.js

```diff
@@ -0,0 +1,67 @@
+const path = require('path');
+
+const CAREER_DETAILS_TEMPLATE_PATH = path.resolve(
+  'src/templates/career-details.tsx',
+);
+
+/**
+ * There can be multiple positions with the same jobId but a different language.
+ * This methods helps to find the (first) position with the same jobId as the given jobId.
+ */
+const createComplementFinder = (positions) => {
+  const groupedPositions = positions.reduce((memo, job) => {
+    memo[job.jobId] = memo[job.jobId] ?? [];
+    memo[job.jobId].push(job);
+    return memo;
+  }, {});
+
+  return {
+    get: (position) =>
+      groupedPositions[position.jobId].find((job) => job.id !== position.id) ??
+      null,
+  };
+};
```

Comment:
This is basically memozing what jobs have the same job id and then we have an easy time to find the other part. By currying we don't need to recalculate the memo everytime. Super premature optimization but I guess my AoC fever got me hire. 

I love how it reads later where it's used 🙏

```
 const complementFinder = createComplementFinder(positions);

  positions.forEach((position) => {
    const complementPosition = complementFinder.get(position);
```